File: Week2 Assignment/add_two_polynomials.c
//Add two polynomials

#include <stdio.h>

#define MAX 100

int main() {
    int deg1, deg2, i;
    int poly1[MAX], poly2[MAX], result[MAX];

    printf("Enter degree of first polynomial: ");
    scanf("%d", &deg1);
    printf("Enter coefficients of first polynomial:\n");
    for (i = 0; i <= deg1; i++) {
        printf("Coefficient of x^%d: ", deg1 - i);
        scanf("%d", &poly1[i]);
    }

    printf("Enter degree of second polynomial: ");
    scanf("%d", &deg2);
    printf("Enter coefficients of second polynomial:\n");
    for (i = 0; i <= deg2; i++) {
        printf("Coefficient of x^%d: ", deg2 - i);
        scanf("%d", &poly2[i]);
    }

    int maxDeg = (deg1 > deg2) ? deg1 : deg2;
    int diff = deg1 - deg2;

    for (i = 0; i <= maxDeg; i++) {
        if (i < diff) {
            result[i] = (deg1 > deg2) ? poly1[i] : poly2[i];
        } else {
            result[i] = poly1[i - (diff > 0 ? 0 : diff)] + poly2[i - (diff < 0 ? 0 : -diff)];
        }
    }

    printf("Resultant polynomial: ");
    for (i = 0; i <= maxDeg; i++) {
        printf("%dx^%d", result[i], maxDeg - i);
        if (i != maxDeg) printf(" + ");
    }

    return 0;
}

File: Week2 Assignment/add_two_polynomials.py
#Add two polynomials

def add_polynomials(poly1, poly2):
    result = poly1.copy()  # Start with the first polynomial
    for power, coeff in poly2.items():
        if power in result:
            result[power] += coeff
        else:
            result[power] = coeff
    return result

def display_polynomial(poly):
    terms = []
    for power in sorted(poly.keys(), reverse=True):
        coeff = poly[power]
        if coeff != 0:
            term = f"{coeff}x^{power}" if power != 0 else f"{coeff}"
            terms.append(term)
    return " + ".join(terms) if terms else "0"


poly1 = {2: 3, 1: 5, 0: 6}
poly2 = {2: 6, 1: 3, 0: 2}

result = add_polynomials(poly1, poly2)

print("Polynomial 1:", display_polynomial(poly1))
print("Polynomial 2:", display_polynomial(poly2))
print("Sum:", display_polynomial(result))
